pkg/operator/controller/route-status/controller_test.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/openshift/cluster-ingress-operator/test/unit"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+)
+
+// Test_Reconcile verifies Reconcile
+// Note: This effectively tests clearStaleRouteAdmittedStatus as well
+func Test_Reconcile(t *testing.T) {
+	reconcileRequestRoute := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "route",
+			Namespace: "ns",
+		},
+	}
+	testCases := []struct {
+		name              string
+		request           reconcile.Request
+		route             *routev1.Route
+		ingressController *v12.IngressController
+		namespace         *corev1.Namespace
+		expectedRoute     *routev1.Route
+		expectedErr       bool
+	}{
+		{
+			name:              "admitted with no routeSelector and no namespaceSelector with route with no labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "admitted with no routeSelector and no namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "admitted with no routeSelector and no namespaceSelector with route with no labels in namespace with correct labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", "shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "admitted with routeSelector and no namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "shard-label", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "shard-label", "", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
```

Comment:
Why did the route's label go away?  I guess it doesn't matter because the test checks the status, not the labels, but it looks funny.